packages/react-reconciler/src/ReactFiberHostContext.js

```diff
@@ -126,14 +126,19 @@ function popHostContext(fiber: Fiber): void {
     // should be fine.
     pop(hostTransitionProviderCursor, fiber);
 
-    // When popping the transition provider, we reset the context value back
-    // to `NotPendingTransition`. We can do this because you're not allowed to nest forms. If
-    // we allowed for multiple nested host transition providers, then we'd
-    // need to reset this to the parent provider's status.
-    if (isPrimaryRenderer) {
-      HostTransitionContext._currentValue = NotPendingTransition;
-    } else {
-      HostTransitionContext._currentValue2 = NotPendingTransition;
+    // When popping the transition provider, we make sure that the entire queue of
+    // the current transition is completed. This is necessary because popping is also
+    // executed when the state changes for children subscribed to the transition provider
+    if (!fiber.memoizedState.memoizedState.pending) {
```

Comment:
Thank you for working on the fix. This introduces another regression though. Since we no longer pop the value if this form is pending, any other form is switched into a pending state now:

https://github.com/user-attachments/assets/8cff14be-3173-4583-b69a-39f85c7c0373

-- https://codesandbox.io/p/sandbox/festive-kalam-5wjrg5


My initial reaction was that we should set the value in pushContext but there's a comment explaining why we can't: https://github.com/facebook/react/blob/6d3c28ca6384351ee34692bfb5d466adf3941962/packages/react-reconciler/src/ReactFiberBeginWork.js#L1937-L1953

Reply:
I think that fixes this particular problem: https://github.com/facebook/react/pull/34075/commits/9be8342214da43906e229d8d5d1547c090a220a9

Reply:
Thanks for taking a look. I will close this in favor of your PR.